splink/comparison_template_library.py

```diff
@@ -0,0 +1,173 @@
+                "label_for_charts": "Matching and 1st Jan",
+            }
+            if m_probability_1st_january:
+                level_dict["m_probability"] = m_probability_1st_january
+            if term_frequency_adjustments:
+                level_dict["tf_adjustment_column"] = col_name
+            comparison_levels.append(level_dict)
+
+        if include_exact_match_level:
+            level_dict = self._exact_match_level(
+                col_name,
+                term_frequency_adjustments=term_frequency_adjustments,
+                m_probability=m_probability_exact_match,
+            )
+            comparison_levels.append(level_dict)
+
+        if len(levenshtein_thresholds) > 0:
+            levenshtein_thresholds = ensure_is_iterable(levenshtein_thresholds)
+
+            if m_probability_or_probabilities_lev is None:
+                m_probability_or_probabilities_lev = [None] * len(
+                    levenshtein_thresholds
+                )
+            ensure_is_iterable(m_probability_or_probabilities_lev)
+
+            for thres, m_prob in zip(
+                levenshtein_thresholds, m_probability_or_probabilities_lev
+            ):
+                level_dict = self._levenshtein_level(
+                    col_name,
+                    distance_threshold=thres,
+                    m_probability=m_prob,
+                )
+                comparison_levels.append(level_dict)
+
+        if len(datediff_thresholds[0]) > 0:
+            datediff_thresholds = ensure_is_iterable(datediff_thresholds)
+
+            if m_probability_or_probabilities_datediff is None:
+                m_probability_or_probabilities_datediff = [None] * len(
+                    datediff_thresholds
+                )
+            ensure_is_iterable(m_probability_or_probabilities_datediff)
+
+            for thres, metric, m_prob in zip(
+                datediff_thresholds[0],
+                datediff_thresholds[1],
+                m_probability_or_probabilities_datediff,
+            ):
+                level_dict = self._datediff_level(
+                    col_name,
+                    date_metric=metric,
+                    date_threshold=thres,
+                    m_probability=m_prob,
+                )
+                comparison_levels.append(level_dict)
+
+            comparison_levels.append(
+                self._else_level(m_probability=m_probability_else),
+            )
+
+            comparison_desc = ""
+            if include_exact_match_level:
+                comparison_desc += "Exact match vs. "
+
+            if len(levenshtein_thresholds) > 0:
+                lev_desc = ", ".join([str(d) for d in levenshtein_thresholds])
+                plural = "" if len(levenshtein_thresholds) == 1 else "s"
+                comparison_desc += (
+                    f"Dates within Levenshtein threshold{plural} {lev_desc} vs. "
+                )
+
+        if len(datediff_thresholds[0]) > 0:
```

Comment:
I'm not sure how I feel about this syntax. See earlier point in the function arguments.

Reply:
I think this is solved now by having separate arguments